hotdoc/core/formatter.py

```diff
@@ -737,8 +737,8 @@ def _format_page(self, page):
 
         out = template.render(
             {'page': page,
-             'scripts': scripts_basenames,
-             'stylesheets': stylesheets_basenames,
+             'scripts': sorted(scripts_basenames),
```

Comment:
Maybe? I tried sorting those too and IIRC it was better but still not perfect.

OrderedSet merely remembers insertion order, I didn't track down where things were getting inserted in non-deterministic order because it was simpler to guarantee they are sorted at time of use.

Also wrapping every `os.listdir()` with `sorted()` resulted in one more sort being done, than doing it at time of use.

Reply:
@eli-schwartz sorry forgot to answer, insertion order can matter I'm afraid, for example one script may depend on another having been loaded first. Going around the various extensions, I can imagine this being a problem with the syntax highlighting extension: <https://github.com/hotdoc/hotdoc/blob/master/hotdoc/extensions/syntax_highlighting/syntax_extension.py#L67-L73> , <https://github.com/hotdoc/hotdoc/blob/master/hotdoc/extensions/syntax_highlighting/prism_autoloader_path_override.js> .

Can you link an example spurious diff?

Reply:
https://github.com/mesonbuild/mesonbuild.github.io/commit/8211d6b3ca644e38ba938b32e8f4a7e1bd2d41c3

assets/js/trie.js keeps getting randomly moved from the top to the bottom and back again... maybe it's grouped by extensions? Possibly the extensions are getting run in random order, but once inserted they are order-preserving.

I haven't yet figured out where that might be taking effect, if at all.